Handle special cases of trigonometric functions.

jsuyash1514 opened the pull request:
This PR adds the results for the corner cases of trigonometric functions sin, cos, and tan.
Solves #1816 
@ewpatton Please have a look.

AppInventorWorkerBee commented:
Can one of the admins verify this patch?

ewpatton commented:
@AppInventorWorkerBee ok to test

jsuyash1514 commented:
@ewpatton I've added some negative values in the tests. Please have a look.

halatmit commented:
I'm not following this carefully, but it seems that the tests are testing
only integer arguments.

Are we also testing that identities are obeyed, e.g., is tan(x)*cos(x)=
sin(x) for all x?


== Hal

Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.



On Fri, Jan 31, 2020 at 3:37 PM Evan W. Patton <notifications@github.com>
wrote:

> *@ewpatton* commented on this pull request.
>
> @jsuyash1514 <https://github.com/jsuyash1514> It might also make sense to
> add some negative values to make sure the modulo math is handling that
> correctly.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/2004?email_source=notifications&email_token=AAGAXNYBH2CULPEHBHTYBG3RASDYHA5CNFSM4KNDGS3KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCT3XU2Y#pullrequestreview-351763051>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGAXN2S7YJOJHCOBPIR56LRASDYHANCNFSM4KNDGS3A>
> .
>

jsuyash1514 commented:
@ewpatton @halatmit I've added a test for a trigonometric identity tan(x)*cos(x) = sin(x). Please let me know if the way I've implemented this test is correct. If so, I would like to add some more trigonometric identities and equations.

ewpatton commented on appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java:
```diff
@@ -1123,6 +1142,24 @@ public void testAtan() throws Throwable {
    }
 
+   public void testTrigonometricIdentities() throws Throwable {
+       Random random = new Random();
```
Indentation is off here. Should be 4 spaces, not 6.

ewpatton commented on appinventor/buildserver/tests/com/google/appinventor/buildserver/YailEvalTest.java:
```diff
@@ -13,6 +13,9 @@
 
+import java.util.Random;
+
+
```
Google style guide has the imports placed alphabetically, so the java import should come after gnu and before junit. You also don't need the extra blank lines.

ewpatton commented on appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm:
```diff
@@ -1793,13 +1793,31 @@
 
 (define (sin-degrees degrees)
-  (sin (degrees->radians-internal degrees)))
+  (if(= (modulo degrees 90) 0)
```
Minor code style thing: I think generally it's accepted to put a space before the first `(` in a sequence of `(`, so there should be a space between the `if` and the `(` that comes after it, e.g., `(if (=`. I recommend this change for everywhere it occurs in your change.

jsuyash1514 commented:
@ewpatton I've made the requested changes and resolved the conflicts with the base branch. Kindly review.

jsuyash1514 commented:
@ewpatton I've added tests for some more trigonometric identities and equations. Please have a look and let me know if anything remains to be done.
